bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/sitequality/SiteQualityPanel.java

```diff
@@ -166,43 +164,59 @@
 				weighty = 50;
 			}});
 
+            final Frame sqpFrame = (Frame) SwingUtilities.getWindowAncestor(this);
+
             add(new Box(BoxLayout.LINE_AXIS) {{
 				add(new JButton(Resources.getIcon("eclipse/add.gif")) {{
                     setToolTipText("Add new timing window");
                     setFocusable(false);
-                    addActionListener(e -> {
-						TimingWindow tw = new TimingWindow();
-						TimingWindowDialog twd = new TimingWindowDialog((Frame) SwingUtilities.getWindowAncestor(SiteQualityPanel.this));
-						tw = twd.showEdit(tw);
-						if (tw != null) {
+                    addActionListener(e ->
+						new TimingWindowDialog(sqpFrame).openNew().foreach(tw -> {
 							owner.getDataObject().addTimingWindow(tw);
 							int index = table.getModel().getRowCount() - 1;
-							table.changeSelection(index, 0, false, false); // magic
-						}
-					});
+							table.changeSelection(index, 0, false, false);
+						}));
                     ButtonFlattener.flatten(this);
 				}});
 
+                add(new JButton(Resources.getIcon("eclipse/download.gif")) {{
+                    setToolTipText("Import timing windows from text file");
+                    setFocusable(false);
+                    addActionListener(e -> {
+                        final TimingWindowImporter importer = new TimingWindowImporter(SiteQualityPanel.this);
+						final TimingWindowParser.TimingWindowParseResults results = importer.openImport();
+
+                        final int newIndex = table.getModel().getRowCount();
+                        owner.getDataObject().addTimingWindows(results.successesAsJava());
+                        if (newIndex < table.getModel().getRowCount())
+                            table.changeSelection(newIndex, 0, false, false);
+
+                        if (results.failures().nonEmpty())
+                            new TimingWindowParseFailureDialog(sqpFrame, results.failures()).setVisible(true);
+                    });
+                    ButtonFlattener.flatten(this);
+                }});
+
                 add(new JButton(Resources.getIcon("eclipse/remove.gif")) {{
                     setToolTipText("Remove timing window");
                     setEnabled(false);
                     setFocusable(false);
 					table.getSelectionModel().addListSelectionListener(e -> setEnabled(table.getSelectedRowCount() > 0));
```

Comment:
I hadn't noticed this case, but good eye. (I tried but missed catching this exception, I guess 😺 .)
I'll see if I can go back through the files and force space indenting instead of tab indenting, since this appears normal in my IDEA but looks hokey here and will probably end up annoying at some point to someone.